### PR TITLE
Move regex to a global variable to not compile it every time

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Here the check is in a for loop so we are compiling the regex every time. Moving it to a global variable makes it faster.

### CODE REVIEW

File: pkg/cgroups/cgroups.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ var (
 
 	// Only mount a single instance
 	cgrpMountOnce sync.Once
+
+	// Regex to match 'kubelet'
+	re = regexp.MustCompile(`^.*kubelet`)
```
Comment (fristonio) [inline pkg/cgroups/cgroups.go]: Can we have a more specific name for this, for instance `kubeletRegex`?

**Review by fristonio — APPROVED** [review]
Thanks for the Pull Request.
Looks Good to me, just a minor nit.

File: pkg/cgroups/cgroups.go (hunk 1 above)
Comment (yeya24) [reply pkg/cgroups/cgroups.go]: Done. PTAL

**Review by christarazi — CHANGES_REQUESTED** [review]
Thanks for the PR!

Two things:

1) Could you squash the commits into one?
2) Could you add reference to the commit that this commit fixes ala step 5 from https://docs.cilium.io/en/v1.9/contributing/development/contributing_guide/#submit-pr?

**Comment by yeya24** [comment]
> Thanks for the PR!
> 
> Two things:
> 
> 1. Could you squash the commits into one?
> 2. Could you add reference to the commit that this commit fixes ala step 5 from https://docs.cilium.io/en/v1.9/contributing/development/contributing_guide/#submit-pr?

Thanks for the review. I have squashed the commits. But for the second point, this pr doesn't fix any bug IMO, it is just a small enhancement.

**Review by christarazi — APPROVED** [review]

**Comment by christarazi** [comment]
test-me-please

File: pkg/cgroups/cgroups.go [hunk 2]
```diff
@@ -96,9 +99,8 @@ func getCgroupRootForKind() (string, error) {
 		}
 
 		if strings.HasPrefix(line, "0::") { // cgroup v2 starts with "0::"
-			re := regexp.MustCompile(`^.*kubelet`)
```
Comment (brb) [inline pkg/cgroups/cgroups.go:99]: There will be only one instace of cgroup v2, so the regexp will be compiled only once in the for loop.

Comment (yeya24) [reply pkg/cgroups/cgroups.go:99]: OK. Sorry I didn't know this before. I will close this pr.

**Pull request closed without merging**